platforms/software/build-init/src/main/java/org/gradle/buildinit/plugins/internal/BuildInitComments.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.buildinit.plugins.internal;
+
+import org.gradle.api.NonNullApi;
+
+@NonNullApi
+public enum BuildInitComments {
+    ON,
```

Comment:
❓ Can you explain the `ON` vs `EXTERNAL` part?

Reply:
`ON` is the default behavior currently used whenever you run `gradle init`.

`EXTERNAL` is a special mode that already existed, and as far as I can tell, it's for our own `gradle/gradle` needs. It is currently used in the `SamplesGenerator`:
https://github.com/gradle/gradle/blob/626bf2eddf23f9f9a407ceb3f10822562af8a60e/build-logic/build-init-samples/src/main/kotlin/gradlebuild/samples/SamplesGenerator.kt#L54-L55

In this mode, instead of writing the actual comments into the source files, only comments like `// <1>` are written, and they are matched later in the `SamplesGenerator` logic of producing a readme.

I would gladly get rid of this mode, but this should be done separately.